packages/@tinacms/api-git/src/router.ts

```diff
@@ -80,7 +94,13 @@ export function router(config: GitRouterConfig = {}) {
       console.log(fileAbsolutePath)
     }
     try {
-      writeFile(fileAbsolutePath, req.body.content)
+      const fileIsInRepo = checkFilePathIsInGitRepo(
+        fileAbsolutePath,
+        REPO_ABSOLUTE_PATH
+      )
+      if (fileIsInRepo) {
+        writeFile(fileAbsolutePath, req.body.content)
+      }
```

Comment:
Right now this operation silently fails. It should instead throw an error. Something like:

```suggestion
      } else {
        throw new Error(`Failed to write to: ${fileRelativePath} \nCannot write outside of the content directory.`)
      }
```